sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientOptions.java

```diff
@@ -654,6 +656,23 @@ IdentityClientOptions setPerRetryPolicies(List<HttpPipelinePolicy> perRetryPolic
         return this;
     }
 
+    /**
+     * Disable instance discovery.
+     * @return the updated client options
+     */
+    public IdentityClientOptions disableInstanceDisovery() {
```

Comment:
I know the naming and usage here is more convenient, but I've never been a fan of this style as its not revertible, so if you can this you can never re-enable instance discovery.

Reply:
Modifying a builder to produce a second credential seems unlikely. At this point I'm going to leave this as is for consistency with other on-by-default options in the Identity SDK.